packages/paypal-commerce-integration/src/paypal-commerce-alternative-methods/paypal-commerce-alternative-method-ratepay-payment-strategy.ts

```diff
@@ -0,0 +1,233 @@
+                ratePay,
+            );
+        }
+    }
+
+    finalize(): Promise<void> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(): Promise<void> {
+        const { legalTextContainer } = this.paypalcommerceratepay || {};
+        if (legalTextContainer) {
+            const legalTextContainerElement = document.getElementById(legalTextContainer);
+            legalTextContainerElement?.remove();
+        }
+
+        return Promise.resolve();
+    }
+
+    private normalizeDate(date: BirthDate) {
+        const formattedDate =
+            date.getDate() < 10 ? `0${date.getDate()}` : date.getDate().toString();
+        const formattedMonth =
+            date.getMonth() < 10 ? `0${date.getMonth() + 1}` : date.getMonth().toString();
+
+        return `${date.getFullYear()}-${formattedMonth}-${formattedDate}`;
+    }
+
+    private renderLegalText() {
+        if (this.paypalcommerceratepay) {
+            const legalTextContainerId = this.paypalcommerceratepay.legalTextContainer;
+            const { container } = this.paypalcommerceratepay;
+            const buttonContainerId = container.split('#')[1];
+            const buttonContainer = document.getElementById(buttonContainerId);
+            const buttonContainerParent = buttonContainer?.parentNode;
+            const legalTextContainer = document.createElement('div');
+            legalTextContainer.style.marginBottom = '20px';
+            legalTextContainer.setAttribute('id', legalTextContainerId);
+            buttonContainerParent?.prepend(legalTextContainer);
+            const paypalSdk = this.paypalCommerceIntegrationService.getPayPalSdkOrThrow();
+            const ratePayButton = paypalSdk.Legal({
+                fundingSource: paypalSdk.Legal.FUNDING.PAY_UPON_INVOICE,
+            });
+            const legalTextContainerElement = document.getElementById(legalTextContainerId);
+
+            if (legalTextContainerElement) {
+                ratePayButton.render(`#${legalTextContainerId}`);
+            }
+        }
+    }
+    private handleError(error: unknown): void {
+        const { onError } = this.paypalcommerceratepay || {};
+        this.toggleLoadingIndicator(false);
+
+        if (onError && typeof onError === 'function') {
+            onError(error);
+        }
+    }
+
+    private createFraudNetScript(merchantId: string, methodId: string, gatewayId: string) {
+        const state = this.paymentIntegrationService.getState();
+        const paymentMethod = state.getPaymentMethodOrThrow<PayPalCommerceInitializationData>(
+            methodId,
+            gatewayId,
+        );
+        const { testMode } = paymentMethod.config;
+        const scriptElement = document.createElement('script');
+        scriptElement.setAttribute('type', 'application/json');
+        scriptElement.setAttribute('fncls', 'fnparams-dede7cc5-15fd-4c75-a9f4-36c430ee3a99');
+        this.guid = this.generateGUID();
```

Comment:
Why did you add your own method for this? As far as i understood it is for generate unique id, if so you can use **uniqueId** from lodash

Reply:
fixed